metabase/metabase
# gate snippet folders and their permissions behind `snippet_collections` premium feature

Closes https://github.com/metabase/metabase-private/issues/76

### Description

Snippet folders and their permissions were not probably gated behind premium feature flags. This PR fixes that according to [our plan](https://www.notion.so/metabase/July-23-Unraveling-of-Feature-Flags-6300baadb8f54fadace6cf0e2fc11a2b).

### How to verify

1. In EE with a token that has the `snippet_collections` feature, Create a SQL question
2. Open the SQL snippets sidebar and create one.
3. Create a snippet folder and move the snippet inside of there.
4. Confirm that you can change permissions at the root level and the folder.
5. Go to admin settings and remove the token.
6. Go back to the snippets sidebar, the snippet should be displayed without any folders, and the permissions options should not be visible.
7. Same should be true on OSS.

### Demo


https://github.com/metabase/metabase/assets/37751258/c82e4e9f-89a0-4e71-bfb6-14f575ef8d2d

EE with token


https://github.com/metabase/metabase/assets/37751258/69a8fe92-d48f-4dd1-9051-4a19fc0e8765

EE without token / OSS

### Checklist
